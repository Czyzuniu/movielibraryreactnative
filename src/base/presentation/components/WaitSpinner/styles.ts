import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  loading: {
    flex: 1,
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    backgroundColor: 'rgba(255, 255, 255, 0.8)',
  },
});
