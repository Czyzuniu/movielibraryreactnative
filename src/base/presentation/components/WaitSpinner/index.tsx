import React from 'react';
import {Center, Spinner, useTheme} from 'native-base';
import {styles} from './styles';

type Props = {
  isVisible?: boolean;
};

export default function WaitSpinner({isVisible = false}: Props) {
  const {
    colors
  } = useTheme();

  return isVisible ? (
    <Center testID={'WAIT_SPINNER'} style={[styles.loading, { backgroundColor: colors.muted['800']}]}>
      <Spinner color="green.500" size={'lg'} />
    </Center>
  ) : null;
}
