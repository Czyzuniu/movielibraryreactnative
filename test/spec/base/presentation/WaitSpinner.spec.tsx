import {render} from '@testing-library/react-native';
import React from 'react';
import ContextWrapper from '../../../utils/ContextWrapper';
import WaitSpinner from '../../../../src/base/presentation/components/WaitSpinner';

describe('WaitSpinner Tests', () => {
  it('should render the spinner if props is visible and match snapshot', () => {
    const {toJSON, getByTestId} = render(
      <ContextWrapper>
        <WaitSpinner isVisible={true} />
      </ContextWrapper>,
    );
    expect(toJSON()).toMatchSnapshot();
    expect(getByTestId('WAIT_SPINNER')).toBeTruthy();
  });

  it('should not render the spinner if props is visible', () => {
    const {queryByTestId} = render(
      <ContextWrapper>
        <WaitSpinner isVisible={false} />
      </ContextWrapper>,
    );
    expect(queryByTestId('WAIT_SPINNER')).toBeFalsy();
  });

  it('should default the prop to not visible', () => {
    const {queryByTestId} = render(
      <ContextWrapper>
        <WaitSpinner />
      </ContextWrapper>,
    );
    expect(queryByTestId('WAIT_SPINNER')).toBeFalsy();
  });
});
