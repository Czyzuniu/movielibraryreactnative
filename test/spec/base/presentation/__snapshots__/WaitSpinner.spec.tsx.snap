// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaitSpinner Tests should render the spinner if props is visible and match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.8)",
          "bottom": 0,
          "flex": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
    testID="WAIT_SPINNER"
  >
    <ActivityIndicator
      accessibilityLabel="loading"
      accessible={true}
      color="#22c55e"
      size="large"
      style={
        Array [
          Array [
            Object {},
            Object {
              "dataSet": Object {},
            },
          ],
          undefined,
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
